### ISSUE TITLE
[Bug]: NoMethodError: undefined method `shop' for nil:NilClass

### ISSUE THREAD

**Issue by rkaucorp** [Shopify/shopify-cli#2125]
### Please confirm that you have:

- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

it should have worked

### Actual behavior

it did not work.

### Stack trace

```shell
/var/lib/gems/2.7.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/dev_server.rb:83:in `rescue in start'
/var/lib/gems/2.7.0/gems/shopify-cli-2.13.0/lib/shopify_cli/theme/dev_server.rb:29:in `start'
/var/lib/gems/2.7.0/gems/shopify-cli-2.13.0/lib/project_types/theme/commands/serve.rb:25:in `call'
/var/lib/gems/2.7.0/gems/shopify-cli-2.13.0/lib/shopify_cli/command/sub_command.rb:16:in `call'
/var/lib/gems/2.7.0/gems/shopify-cli-2.13.0/lib/shopify_cli/command.rb:27:in `call'
```

### Reproduction steps

1.
2.
3.

### Operating System

x86_64-linux-gnu

### CLI version (run `shopify version` if you're not sure)

2.13.0

### Shell

/usr/bin/zsh

### Ruby version (run `ruby -v` if you're not sure)

2.7.0p0

**Comment by DanielRiveraHQ** [comment]
I have the same issue, is there any update?

**Comment by Slgoetz** [comment]
Same issue - Any update?

**Comment by pmourer-zaneray** [comment]
this problem is referenced in at least one other issue when I search. if that issue does not describe your problem, you should update this issue to more accurately describe your problem and the differences with other issues.

https://github.com/Shopify/shopify-cli/issues?q=is%3Aissue+is%3Aopen+nilclass

**Comment by karreiro** [comment]
Thanks for adding me to the loop and for the thoughts, @amcaplan.

The development theme cannot be created when any permission error happens. So, that error was occurring.

That scenario is already fixed by #2114 tho (released yesterday).

So, I'm closing this one in favor of #2059.

**Comment by amcaplan** [comment]
Even better, thanks @karreiro!
